modules/core/src/space/space-object-base.ts

```diff
@@ -47,6 +47,10 @@ export abstract class SpaceObjectBase extends Schema {
 
     @gameField('string')
     public id = '';
+
+    @gameField({ map: 'uint8' })
+    public scanLevels = new MapSchema<number>(); // Key: faction name, Value: 0-2 (ScanLevel)
```

Comment:
If the key is an enum, the data structure should not be dynamic, that's a waste of performance and complexity. use an array that is initialized to have the minimal scan level for every faction.